### PR TITLE
Module2 hw3

### PR DESCRIPTION
**Pull request by Joker3383**

### CODE REVIEW

File: HomeworkThree/HomeworkThree/Models/Vehicle.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            Speed = speed;
+            Weight = weight;
+
+        }
+
+        public virtual void Move(float speed) => Console.WriteLine($"This transport moves on with {speed}km/h");
+
+        public int CompareTo(Vehicle? other)
+        {
+            return GetSpeed().CompareTo(other!.GetSpeed());
```
Comment (YevhenBuriak) [inline HomeworkThree/HomeworkThree/Models/Vehicle.cs:23]: Better to avoid !. operator

File: HomeworkThree/HomeworkThree/Models/Vehicle.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿namespace HomeworkThree
+{
+    public class Vehicle : IMoveable, IComparable<Vehicle>
```
Comment (YevhenBuriak) [inline HomeworkThree/HomeworkThree/Models/Vehicle.cs:3]: I assume users of your code can create just 'vehicle'?

File: HomeworkThree/HomeworkThree/Models/Car.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace HomeworkThree
+{
+    public class Car : Vehicle 
+    {
+        public bool IsRaces { get; set; }
+        public Car(string name, int year, float speed, float weight,bool isRaces) :base(name,year,speed,weight)
+        {
+            IsRaces = isRaces;
+        }
```
Comment (YevhenBuriak) [inline HomeworkThree/HomeworkThree/Models/Car.cs:15]: Formatting

File: HomeworkThree/HomeworkThree/Models/Car.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        public virtual void IsConfortable(bool IsRaces)
```
Comment (YevhenBuriak) [inline HomeworkThree/HomeworkThree/Models/Car.cs:16]: params namig

File: HomeworkThree/HomeworkThree/Models/Vehicle.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+        public virtual void Move(float speed) => Console.WriteLine($"This transport moves on with {speed}km/h");
+
+        public int CompareTo(Vehicle? other)
+        {
+            return GetSpeed().CompareTo(other!.GetSpeed());
+        }
+        public virtual double GetSpeed()
+        {
+            return 0.0;
```
Comment (YevhenBuriak) [inline HomeworkThree/HomeworkThree/Models/Vehicle.cs:27]: Why default returns 0 and not value of the Speed property?

**Review by YevhenBuriak — APPROVED** [review]

File: HomeworkThree/HomeworkThree/Services/SearchName.cs [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace HomeworkThree
+{
+    public static class SearchName
+    {
+        public static void Search(Dictionary<string,Vehicle> dict)
```
[the remark this answers is not in the document]
Comment (Joker3383) [reply HomeworkThree/HomeworkThree/Services/SearchName.cs:11]: i forgot used 'this' in the parameters ?

**Pull request merged**